Bottom: 8167bbec96d5ece38d6a5982c2b71295fae5276d
Top:    6e371006a266e270caa39cb492a9437ba3bb4a73
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-25 22:11:16 +0000

Normalise plane reference directions emitted

This fixes our rendering of our own exported models. Probably we
should also normalise direction vectors as we load the board.
Most CAD software seems to cope with the non-normalised directions
so we probably ought to as well.


---

diff --git a/src/hid/step/object3d_step.c b/src/hid/step/object3d_step.c
index eeb73e0..213607e 100644
--- a/src/hid/step/object3d_step.c
+++ b/src/hid/step/object3d_step.c
@@ -198,6 +198,7 @@ object3d_to_step_body_fragment (step_file *step,
           vertex3d *dv = DDATA (outer_contour->first_edge);
 
           double rx, ry, rz;
+          double length;
 
           rx = dv->x - ov->x;
           ry = dv->y - ov->y;
@@ -211,6 +212,11 @@ object3d_to_step_body_fragment (step_file *step,
               rx = 1., ry = 0., rz = 0.;
             }
 
+          length = hypot (hypot (rx, ry), rz);
+          rx /= length;
+          ry /= length;
+          rz /= length;
+
           face->surface_identifier =
             step_plane (step, "NONE",
                         step_axis2_placement_3d (step, "NONE",
